ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -1,21 +1,82 @@
-import asyncio
+import functools
 
 from ..asyncio import context_provider
 from ...compat import stringify
 from ...constants import ANALYTICS_SAMPLE_RATE_KEY, SPAN_MEASURED_KEY
 from ...ext import SpanTypes, http
 from ...propagation.http import HTTPPropagator
 from ...settings import config
+from ...context import Context
 
+try:
+    from aiohttp.web import middleware
 
-CONFIG_KEY = 'datadog_trace'
-REQUEST_CONTEXT_KEY = 'datadog_context'
-REQUEST_CONFIG_KEY = '__datadog_trace_config'
-REQUEST_SPAN_KEY = '__datadog_request_span'
+    AIOHTTP_HAS_MIDDLEWARE = True
+except ImportError:
+    AIOHTTP_HAS_MIDDLEWARE = False
 
+    def middleware(f):
+        return f
 
-@asyncio.coroutine
-def trace_middleware(app, handler):
+
+CONFIG_KEY = "datadog_trace"
+REQUEST_CONTEXT_KEY = "datadog_context"
+REQUEST_CONFIG_KEY = "__datadog_trace_config"
+REQUEST_SPAN_KEY = "__datadog_request_span"
+
+propagator = HTTPPropagator()
+
+
+config._add("aiohttp_server", dict(
```

Comment:
should we instead make this two level?

ex:  `conf.add("aiohttp", dict(server=dict(...), client=(...))`. otherwise is there any reason for using `_` instead of `.` ?